uber/models/email.py

```diff
@@ -64,3 +65,27 @@ def html(self):
             return safe_string(body)
         else:
             return safe_string(self.body.replace('\n', '<br/>'))
+
+
+class EmailDaemonStatus(MagModel):
+    last_run_finished = Column(UTCDateTime, server_default=utcnow())
+
+    @staticmethod
+    def last_result(session):
+        status = session.query(EmailDaemonStatus).all()
+        assert len(status) <= 1, "there should only ever be at most 1 row in this table"
+        return status[0] if len(status) == 1 else None
+
+    @staticmethod
+    def last_result_looks_valid(session):
+        # return the result if the email server has completed a run in the past day
+        last_run = EmailDaemonStatus.last_result(session)
+        if not last_run:
+            return None
+
+        return last_run if last_run.last_run_finished > datetime.now(UTC) - timedelta(days=1) else None
+
+
+class EmailDaemonCategoryResult(MagModel):
+    ident = Column(UnicodeText)
+    unsent_because_unapproved = Column(Integer, default=0)
```

Comment:
I specifically didn't try linking up foreign keys to the AutomatedEmails category here because I wanted this data to be a bit less tightly coupled and more like reading a report.  If keys/idents change, I wanted to make the client code tolerant of those changes without having to worry about it too much.

Reply:
I agree that the lack of coupling is a good idea in this case.  It'll probably make future refactors easier this way.